PROVIDER_CHECKLIST.md

```diff
@@ -0,0 +1,48 @@
+# Provider Checklist
+
+Crossplane manages Resources external via [Providers](https://crossplane.io/docs/master/concepts/providers.html). Providers are composed of Kubernetes [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) and [Controllers](https://kubernetes.io/docs/concepts/architecture/controller) that communicate to a remote API and manages the lifecycle of a resource from Creation to Deletion.
+
+To build a new provider, please refer to the  [Provider Development Guide](https://crossplane.io/docs/master/contributing/provider_development_guide.html).
+
+This document contains list of items that are usually involved in managing Open Source contributions to a Crossplane provider. In general your provider should follow the guidelines documented in the [Crossplane Contributing](https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md) guide.
+
+## Repository
+
+It is recommended to use the [crossplane-contrib](https://github.com/orgs/crossplane-contrib) Github Organization. Your project should be named `provider-<name>`. Members of the Crossplane community are happy to create a repository and configure access. Please reach out to the Crossplane [#dev](https://crossplane.slack.com/archives/CEF5N8X08) channel.
```

Comment:
I wonder if we should ask folks to raise an issue on this repo (as well as a Slack ping)? I find Slack pings often get lost when we're busy.

Reply:
What if we made this part of the workflow in the [crossplane/org ](https://github.com/crossplane/org)? We could have a new-repository issue template that includes many parts of this workflow?

Reply:
Opened issue https://github.com/crossplane/org/issues/26